sigsep_open-unmix-pytorch_umx.md

```diff
@@ -0,0 +1,74 @@
+---
+layout: hub_detail
+background-class: hub-background
+body-class: hub
+category: researchers
+title: Open-Unmix
+summary: Reference implementation for music source separation
+image: sigsep_logo_inria.png
+author: Inria
+tags: [audio, scriptable]
+github-link: https://github.com/sigsep/open-unmix-pytorch
+github-id: sigsep/open-unmix-pytorch
+featured_image_1: sigsep_umx-diagram.png
+featured_image_2: no-image
+accelerator: cuda-optional
+---
+
+
+```bash
+# assuming you have a PyTorch >=1.6.0 installed
+pip install -q torchaudio
+```
+
+```python
+import torch
+
+# loading umxhq four target separator
+separator = torch.hub.load('sigsep/open-unmix-pytorch', 'umxhq')
+
+# audio shape (nb_samples, nb_channels, nb_timesteps)
+# assuming audio is resampled to separator.sample_rate
+audio = torch.rand((1, 2, 100000))
+estimates = separator(audio)
+# estimates.shape = (1, 4, 2, 100000)
+```
+
+### Model Description
+
+__Open-Unmix__ provides ready-to-use models that allow users to separate pop music into four stems: __vocals__, __drums__, __bass__ and the remaining __other__ instruments. The models were pre-trained on the freely available [MUSDB18](https://sigsep.github.io/datasets/musdb.html) dataset.
+
+Open-unmix comprises multiple models that are trained for each particular target.
+
+Each target model is based on a three-layer bidirectional deep LSTM. The model learns to predict the magnitude spectrogram of a target source, like vocals, from the magnitude spectrogram of a mixture input. Internally, the prediction is obtained by applying a mask on the input. The model is optimized in the magnitude domain using mean squared error.
+
+A `Separator` meta-model puts together multiple _Open-unmix_ spectrogram models for each desired target, and combines their output through a multichannel generalized Wiener filter, before application of inverse STFTs using `torchaudio`.
```

Comment:
do you have an example for the `Separator` model since it behaves differently?

Reply:
thanks for the comment. Again, this wasn't clear enough on my side....

the example above, already loads a `separator` which is a joint model using wiener filtering. In contrast, users can also load the individual models and use them. 

@vincentqb do you think we should add the following snippet?

```python
# magnitude spectrogram shape (nb_samples, nb_channels, nb_bins, nb_frames)
spectrogram = torch.rand((1, 2, 100, 2049))
models = torch.hub.load('sigsep/open-unmix-pytorch', 'umxhq_spec')
separated_vocal_spectrogram = models['vocals'](spectrogram)
```

Reply:
would this work?
```
separator = torch.hub.load('sigsep/open-unmix-pytorch', 'umxhq')
audio = torch.rand((1, 2, 100000))
estimates_vocals_only = separator["vocals"](audio)
```

Reply:
No, the `separator` are end-to-end time domain models (`umx`, `umxhq`, `umxse`), only `_spec` give access to the individual spectrogram models.

Instead, this is what the `targets` argument is intended for....

e.g. in the following example, only the vocal submodel is used and the wiener filtering is turned off:

```
separator = torch.hub.load('sigsep/open-unmix-pytorch', 'umxhq', targets=['vocals'], niter=0)
audio = torch.rand((1, 2, 100000))
estimates_vocals_only = separator(audio)
```

with wiener filtering, the residual source (`mix-estimate`) would need to be turned out 

```
separator = torch.hub.load('sigsep/open-unmix-pytorch', 'umxhq', targets=['vocals'], residual=True)
audio = torch.rand((1, 2, 100000))
estimates_vocals_only = separator(audio)[:, 0, ...]
```

Happy to add one of these,  what do you think @vincentqb?

Reply:
Actually, now that I re-read the text with what you said, I don't think we need to add another example: instead, I'd just emphasize that the separator in the code example is what's referred to in the last paragraph.